src/main/scala/mesosphere/marathon/core/matcher/base/util/ActorOfferMatcher.scala

```diff
@@ -18,7 +18,7 @@ import scala.concurrent.{ Future, Promise }
   * @param actorRef Reference to actor that matches offers.
   * @param precedenceFor Defines which matcher receives offers first. See [[mesosphere.marathon.core.matcher.base.OfferMatcher.precedenceFor]].
   */
-class ActorOfferMatcher(actorRef: ActorRef, override val precedenceFor: Option[PathId], homeRegionProvider: Provider[HomeRegionProvider])
+class ActorOfferMatcher(actorRef: ActorRef, override val precedenceFor: Option[PathId], homeRegionProvider: () => HomeRegionProvider)
```

Comment:
Do you think it's possible to pass a function here? Something like `homeRegion: () => string`

Reply:
right ... good point. I don't know how I would do it via guice but now it should be pretty straightforward